gradle/validation/ecj-lint.gradle

```
@@ -29,6 +29,51 @@ configure(rootProject) {
 
 def resources = scriptResources(buildscript)
 
+class EcjLintArgumentProvider implements CommandLineArgumentProvider {
+
+    @Input
+    String targetCompatibility
+
+    @InputFiles
+    @PathSensitive(PathSensitivity.RELATIVE)
```

```suggestion
    @Classpath
```

This might make more sense since we only care about the contents of the jars on the classpath.